rxros_teleop/src/VelocityPublisher_rx.cpp

```diff
@@ -77,7 +77,9 @@ int main(int argc, char** argv) {
         else if (event == JS_EVENT_AXIS_LEFT || event == KB_EVENT_LEFT)
             return std::make_tuple(prevVelLinear, adaptVelocity((prevVelAngular + deltaVelAngular), minVelAngular, maxVelAngular, true)); // move left
         else if (event == JS_EVENT_AXIS_RIGHT || event == KB_EVENT_RIGHT)
-            return std::make_tuple(prevVelLinear, adaptVelocity((prevVelAngular - deltaVelAngular), minVelAngular, maxVelAngular, false));}; // move right
+            return std::make_tuple(prevVelLinear, adaptVelocity((prevVelAngular - deltaVelAngular), minVelAngular, maxVelAngular, false)); // move right
+        else
+            return std::make_tuple(prevVelLinear, prevVelAngular);}; // do nothing    
```

Comment:
Is the need for this the fact that there *has* to be some dataflow out of this node, because of how RxCPP works? If there is 'nothing' to do, wouldn't not publishing / creating a message be the right course of action here?

Reply:
There are several issues here:
1. Without the last else statement the lambda function is incomplete. It is uncertain what the function will return if none of the if/else if statements are not met. To some extend I am actually a bit surprised that the compiler does not fail or gives a warning in this case - this is probably some compiler options that are missing here.
2. You may be right that a more clean solution would have been to returned a do_nothing result, although it would require a few more changes. I am actually satisfied with the proposed solution. It requires only a new else statement and the coding style is both functional and easy to understand (read I am lacy :-)